roles/v2/mining-proxy/src/lib/downstream_mining.rs

```diff
@@ -301,10 +331,63 @@ impl
 
     fn handle_open_standard_mining_channel(
         &mut self,
-        _: OpenStandardMiningChannel,
-        up: Option<Arc<Mutex<UpstreamMiningNode>>>,
+        req: OpenStandardMiningChannel,
+        mut up: Option<Arc<Mutex<UpstreamMiningNode>>>,
     ) -> Result<SendTo<UpstreamMiningNode>, Error> {
-        Ok(SendTo::RelaySameMessageToRemote(up.unwrap()))
+        let channel_kind = up
+            .as_ref()
+            .unwrap()
+            .safe_lock(|up| up.channel_kind)
+            .unwrap(); //remove unwrap
```

Comment:
Did you intend on removing this unwrap?

Reply:
First unwrap is now removed and changed with an expect. The second one is fine because on a safe_lock. The comment was for the first one only